### PR TITLE
[AzureMonitorExporter] update demo project

### PR DESCRIPTION
**Pull request by TimothyMothra**
### Changes
- fix nullables
  Towards #34013
- remove extra files for TraceDemo (`ActivityFilteringProcessor` and `ActivityEnrichingProcessor`.
  move these classes into the TraceDemo.cs file.
  This is to improve discoverability of these files, as commented here: https://github.com/Azure/azure-sdk-for-net/pull/31204#discussion_r1095017414
- add OTEL-DIAGNOSTICS.json file to demonstrate how to collect internal logs.
  This is also useful for dev loop to quickly collect logs while debugging.
- misc cleanup; spelling, whitespace.
- update Readme with link to OTel Troubleshooting.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Azure.Monitor.OpenTelemetry.Exporter.Demo.csproj [hunk 1]
```diff
@@ -5,6 +5,16 @@
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Azure.Monitor.OpenTelemetry.Exporter.Demo.csproj:8]: Do we need both of the entries here?

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]
LGTM: Should we update https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter#troubleshooting to point to demo as a hint?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Azure.Monitor.OpenTelemetry.Exporter.Demo.csproj (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Demo/Azure.Monitor.OpenTelemetry.Exporter.Demo.csproj:8]: good catch, thanks!
i'll clean this up!

**Comment by TimothyMothra** [comment]
> LGTM: Should we update https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter#troubleshooting to point to demo as a hint?

Great callout! I'll update the Readme

**Pull request merged**